src/components/shared/ButtonWithPanel.js

```diff
@@ -2,15 +2,43 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React, { PureComponent, PropTypes } from 'react';
+// @flow
+
+import React, { PureComponent } from 'react';
 import classNames from 'classnames';
 
 require('./ButtonWithPanel.css');
 
+interface Panel {
+  props: {
+    onOpen: () => mixed,
+    onClose: () => mixed,
+  },
+  open(): mixed,
+}
+
+type Props = {
+  className: string,
+  label: string,
+  panel: React$Element<*>, // Ideally we'd like to say that panel implements Panel, but I can't express it with Flow
+  open?: boolean,
+};
+
 class ButtonWithPanel extends PureComponent {
-  constructor(props) {
+  props: Props;
+  state: {|
+    open: boolean,
+  |};
+
+  _panel: Panel | null;
+
+  _onPanelOpen: () => void;
+  _onPanelClose: () => void;
+  _panelCreated: Panel => void;
+
+  constructor(props: Props) {
     super(props);
-    this.state = { open: false };
+    this.state = { open: !!props.open };
```

Comment:
I'm curious as to what this change does.

Reply:
The goal is to be able to pass "open" as a prop, instead of calling "openPanel" :) This is a lot more React-y !